woocommerce/woocommerce-ios
# [IAP] Show `plan ended` when IAP plan has expired

Closes: #9915


## Description


In 🍍 we're adding the ability to purchase Woo Express plans using In-App Purchases (IAP).

We've been testing this with Sandboxed IAP accounts, which are configured with an accelerated renewal timeline. After subscribing to a "monthly" plan, we get up to 12 renewals at 5 minute intervals, and then the plan expires/is cancelled.

We've noticed that after this, the site doesn't show any useful information in the Subscriptions view.

Generally, [expiry is handled knowing the specific plan which expired](https://github.com/woocommerce/woocommerce-ios/blob/8132e61139ec62dc01b8bc1250b006e4c2cb0202/WooCommerce/Classes/ViewRelated/Upgrades/SubscriptionsViewModel.swift#L133), however from IAP (at least sandboxed ones) we don’t get told what the expired plan was. (`/sites/{id}/plans` response has no entry with `current_plan:true`.)

In this case, we will show generic messaging so that the merchant knows they don’t have a plan.

## Testing instructions


### Preparation
Create a sandboxed IAP app store user in App Store Connect, and set that as the sandboxed App Store Apple ID in settings. See PdfdoF-1Ej-p2 for more detail on this.

### Test

1. Navigate to Menu > Site selector
2. Scroll down and tap `+ create new store`
3. Go through the store creation flow
4. When complete, tap `Upgrade now` in the banner that shows on the dashboard
5. Complete the IAP to upgrade the store
6. Wait an hour or so
7. Open the app again, and go to Menu > Subscriptions
8. Observe that your app shows `Subscription Status` - `Current: plan ended`

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/2472348/d5eda835-41c1-4dc0-ad3a-e1250499fafe



---
